lib/Doctrine/ODM/CouchDB/UnitOfWork.php

```diff
@@ -291,7 +291,7 @@ private function createDocumentAttachments($documentId, $data)
         $basePath = '/' . $this->dm->getCouchDBClient()->getDatabase() . '/' . $documentId . '/';
         foreach ($data AS $filename => $attachment) {
             if (isset($attachment['stub']) && $attachment['stub']) {
-                $instance = Attachment::createStub($attachment['content_type'], $attachment['length'], $attachment['revpos'], $client, $basePath . $filename);
+                $instance = Attachment::createStub($attachment['content_type'], $attachment['length'], $attachment['revpos'], $client, $basePath . urlencode($filename));
```

Comment:
I am pretty sure this is something that the `Attachment` class should fix, not the calling code. Or?